### PR TITLE
Refactor scripts to always use `oc get --raw` when possible

### PR DESCRIPTION
**Pull request by stevekuznetsov**
When requesting protected resources from the OpenShift server,
previously a set of Bash utilities was used to allow `curl` to
pass the correct cert and key. Now that we have the functionality
to get raw resources from the API server in the OpenShift client,
we don't need to use any of those utilities any longer against the
server.

Requests that were being made against the embedded etcd store
were incorrectly using the master cert and key previously, but
now are amended to correctly use the etcd client cert and key.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

@deads2k @smarterclayton @liggitt @csrwng PTAL

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
[test]

**Comment by smarterclayton** [comment]
This only works because admin.kubeconfig has the same client cert as etcd?

On Oct 31, 2016, at 1:04 PM, Steve Kuznetsov <notifications@github.com>
wrote:

[test]

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/11669#issuecomment-257353895>,
or mute
the thread

File: hack/lib/start.sh [hunk 1]
```diff
@@ -383,7 +383,9 @@ function os::start::etcd() {
 	echo "[INFO] etcd server start at: "
 	date
 
-	wait_for_url "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/version" "etcd: " 0.25 80
+	os::test::junit::declare_suite_start "setup/start-etcd"
+	os::cmd::try_until_success "curl --fail --silent ${API_SCHEME}://${API_HOST}:${ETCD_PORT}/version" "$(( 80 * second ))" "$(( 250 * millisecond ))"
```
Comment (stevekuznetsov) [inline hack/lib/start.sh]: Should we pin a version?

File: hack/util.sh [hunk 2]
```diff
@@ -247,8 +157,7 @@ function cleanup_openshift() {
 	grep -a5 "CACHE.*ALTERED" ${LOG_DIR}/openshift.log
 
 	echo "[INFO] Dumping etcd contents to ${ARTIFACT_DIR}/etcd_dump.json"
-	set_curl_args 0 1
-	curl -s ${clientcert_args} -L "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/v2/keys/?recursive=true" > "${ARTIFACT_DIR}/etcd_dump.json"
+	oc get --raw "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/v2/keys/?recursive=true" --config="${MASTER_CONFIG_DIR}/admin.kubeconfig" > "${ARTIFACT_DIR}/etcd_dump.json"
```
Comment (stevekuznetsov) [inline hack/util.sh]: This won't work, will it. Hmmm.

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -12,8 +12,7 @@ function cleanup()
     set +e
 
     echo "[INFO] Dumping etcd contents to ${ARTIFACT_DIR}/etcd_dump.json"
-    set_curl_args 0 1
-    curl -s ${clientcert_args} -L "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/v2/keys/?recursive=true" > "${ARTIFACT_DIR}/etcd_dump.json"
+    oc get --raw "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/v2/keys/?recursive=true" --config="${MASTER_CONFIG_DIR}/admin.kubeconfig" > "${ARTIFACT_DIR}/etcd_dump.json"
```
Comment (liggitt) [inline hack/test-cmd.sh]: admin.kubeconfig does not contain etcd credentials
master.etcd-client.crt/master.etcd-client.key/ca.crt are more what you want

**Comment by stevekuznetsov** [comment]
@liggitt or @smarterclayton I would appreciate it if you could try this on MacOS

[test] me

**Comment by stevekuznetsov** [comment]
[test] me

**Comment by smarterclayton** [comment]
Lgtm

**Comment by stevekuznetsov** [comment]
Hmmm... the dump only contains 
```json
{"action":"get","node":{"dir":true}}
```

After test-cmd..

**Comment by stevekuznetsov** [comment]
That actually looks like the same as the current behavior on `master` -- @deads2k is that expected?

**Comment by stevekuznetsov** [comment]
Unless @deads2k blocks on that output (which I would prefer to fix in another PR), @smarterclayton this is ready to merge

**Comment by deads2k** [comment]
> That actually looks like the same as the current behavior on master -- @deads2k is that expected?

No.  I would expect the dump to contain resources.  For instance, without PolicyBindings, no one can access the cluster.

**Comment by stevekuznetsov** [comment]
@smarterclayton broke that dump when we migrated to etcd v3 -- will fix in a follow-up. [merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**